little_raft/src/replica.rs

```diff
@@ -377,6 +377,11 @@ where
                 let mut state_machine = self.state_machine.lock().unwrap();
                 state_machine
                     .register_transition_state(transition.get_id(), TransitionState::Queued);
+            } else {
```

Comment:
Looks good to me. Little Raft is single-threaded, so we don't have to worry about state changing under our feet, so let's move the Leader check on line 369 here out of the `for` loop for efficiency just like you did in the previous draft PR.

And then let's reuse `let mut state_machine` declaration by declaring the variable outside the `if` statement.